examples/lessons/08/app.js

```diff
@@ -1,181 +1,230 @@
 /* eslint-disable no-var */
 /* eslint-disable max-statements, array-bracket-spacing, no-multi-spaces */
 /* global window, document */
-import {createGLContext, loadTextures, Cube, Program, addEvents, Fx} from 'luma.gl';
+import {GL, AnimationLoop, Cube, Matrix4, Texture2D,
+  addEvents, loadTextures, Model} from 'luma.gl';
+
+// Vertex shader with lighting
+const VERTEX_SHADER = `\
+attribute vec3 positions;
+attribute vec2 texCoords;
+attribute vec3 normals;
+
+uniform mat4 uMVMatrix;
+uniform mat4 uPMatrix;
+uniform vec3 uAmbientColor;
+uniform vec3 uLightingDirection;
+uniform vec3 uDirectionalColor;
+uniform bool uUseLighting;
+
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
+
+void main(void) {
+  gl_Position = uPMatrix * uMVMatrix * vec4(positions, 1.0);
+  vTextureCoord = texCoords;
+
+  if (!uUseLighting) {
+      vLightWeighting = vec3(1.0, 1.0, 1.0);
+  } else {
+      // Perform lighting in world space
+      // we should use 'transpose(inverse(mat3(uMVMatrix)))', but
+      // 'inverse' matrix operation not supported in GLSL 1.0, for now use
+      // upper-left 3X3 matrix of model view matrix, it works since we are not
+      // doing any non-uniform scaling transormations in this example.
+      mat3 normalMatrix = mat3(uMVMatrix);
+      vec3 transformedNormal = normalMatrix * normals;
+      float directionalLightWeighting = max(dot(transformedNormal, uLightingDirection), 0.0);
+      vLightWeighting = uAmbientColor + uDirectionalColor * directionalLightWeighting;
+  }
+}
+`;
 
-// Blend Fragment Shader
-const blendFS = `\
+// Fragment shader
+const FRAGMENT_SHADER = `\
 #ifdef GL_ES
 precision highp float;
 #endif
 
-varying vec4 vColor;
-varying vec2 vTexCoord;
-varying vec3 lightWeighting;
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
 
-uniform bool hasTexture1;
-uniform sampler2D sampler1;
-uniform float alpha;
+uniform sampler2D uSampler;
+uniform float uAlpha;
 
-void main() {
-  if (hasTexture1) {
-    gl_FragColor = vec4(texture2D(sampler1,
-      vec2(vTexCoord.s, vTexCoord.t)).rgb * lightWeighting, alpha);
-  }
+void main(void) {
+  vec4 textureColor = texture2D(uSampler, vec2(vTextureCoord.s, vTextureCoord.t));
+  gl_FragColor = vec4(textureColor.rgb * vLightWeighting, textureColor.a * uAlpha);
 }
 `;
 
+var xRot = 0;
+var xSpeed = 0.01;
+var yRot = 0;
+var ySpeed = 0.0;
+var cubePositionZ = -5.0;
+
 const animationLoop = new AnimationLoop({
-  onInitialize({gl}) {
-    loadTextures(gl, {
+  onInitialize: ({canvas, gl}) => {
+    addControls();
+    addKeyboardHandler(canvas);
+
+    gl.clearColor(0, 0, 0, 1);
+    gl.clearDepth(1);
+    gl.enable(GL.DEPTH_TEST);
+    gl.depthFunc(GL.LEQUAL);
+    gl.blendFunc(gl.SRC_ALPHA, gl.ONE);
```

Comment:
As I enabled blending in this lesson, now when I switch to other lessons, blending is enabled there too, does all lessons use same GL context?